src/models/propulsion/FGBldc.cpp

```diff
@@ -0,0 +1,224 @@
+      if (el->FindElement("noloadcurrent"))
+          noLoadCurrent = el->FindElementValueAsNumber("noloadcurrent");
+      if (el->FindElement("decelerationTime"))
+          deceleration_time = el->FindElementValueAsNumber("decelerationTime");
+ 
+      MaxCurrent = MaxVolts / coilResistance + noLoadCurrent;
+
+      PowerWatts = MaxCurrent * MaxVolts;
+
+      string base_property_name = CreateIndexedPropertyName("propulsion/engine",
+                                                            EngineNumber);
+      exec->GetPropertyManager()->Tie(base_property_name + "/power-hp", &HP);
+
+      exec->GetPropertyManager()->Tie(base_property_name + "/current-a", &CurrentRequired);
+
+      Debug(0); // Call Debug() routine from constructor if needed
+}
+
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+FGBldc::~FGBldc()
+    {
+      Debug(1); // Call Debug() routine from constructor if needed
+}
+
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+void FGBldc::Calculate(void)
+{
+    RunPreFunctions();
+
+    if (Thruster->GetType() == FGThruster::ttPropeller) {
+        ((FGPropeller*)Thruster)->SetAdvance(in.PropAdvance[EngineNumber]);
+        ((FGPropeller*)Thruster)->SetFeather(in.PropFeather[EngineNumber]);
+    }
+
+    RPM = Thruster->GetRPM();
+    TorqueRequired = abs(((FGPropeller*)Thruster)->GetTorque());
+
+    CurrentRequired = (TorqueRequired * VelocityConstant) / TorqueConstant;
+
+//  total current required must include no load current i0
+    CurrentRequired = CurrentRequired + noLoadCurrent;
+ 
+    V = MaxVolts * in.ThrottlePos[EngineNumber];
+  
+//  Commanded RPM = (input voltage - currentRequired * coil resistance) * velocity costant
+    CommandedRPM = (V - CurrentRequired * coilResistance) * VelocityConstant;
+    DeltaRPM = round((CommandedRPM - RPM));
+
+//  Torque is MaxTorque (stall torque) at 0 RPM and linearly go to 0 at max RPM (MaxVolts*VelocityCostant)
+//  MaxTorque = MaxCurrent*torqueconstant/velocityconstant*(1-RPM/maxRPM)
+
+    MaxTorque = MaxCurrent / VelocityConstant * TorqueConstant * (1 - RPM / (MaxVolts* VelocityConstant));
+
+    TorqueAvailable = MaxTorque - TorqueRequired;
+    DeltaTorque = (((DeltaRPM/60)*(2.0 * M_PI))/(max(0.00001, in.TotalDeltaT))) * ((FGPropeller*)Thruster)->GetIxx();
+ 
+//  compute acceleration and deceleration phases:
+//  Acceleration is due to the max delta torque available and is limited to the inertial forces
+
+    if (DeltaRPM >= 0) {
+      TargetTorque = min(DeltaTorque, TorqueAvailable) + TorqueRequired;
+    } else {
+//  Deceleration is due to braking force given by the ESC and set by parameter deceleration_time 
+      TargetTorque = TorqueRequired  - min(abs(DeltaTorque)/(max(deceleration_time,0.01)*30),RPM*TorqueConstant/VelocityConstant/VelocityConstant/coilResistance);
+    }
+
+    EnginePower = ((2 * M_PI) * max(RPM, 0.0001) * TargetTorque) / 60;
+    HP = EnginePower / 550;
```

Comment:
```suggestion
    HP = EnginePower / hptowatt;
```
EnginePower is in Watt, isn't it ? In which case, the conversion should use `hptowatt`.